src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/PublicSymbols/AnonymousType.DelegatePublicSymbol.cs

```diff
@@ -14,11 +14,13 @@ internal sealed partial class AnonymousTypeManager
     {
         internal sealed class AnonymousDelegatePublicSymbol : AnonymousTypeOrDelegatePublicSymbol
         {
+            internal readonly ImmutableArray<TypeParameterSymbol> ContainingTypeParameters;
```

Comment:
>ContainingTypeParameters [](http://example.com/codeflow?start=66&length=24)

Something feels wrong about this set of type parameters. If we say that this type should have type parameters on its own, then this what we should implement. Another way to somehow express that is to set ```ContainingSymbol``` to be the first generic container that contains the code that creates the type. This feels strange too, because we will be introducing a concept of a type contained within a method. Also, these type parameters feel important, yet they do not participate in equality. The structure of symbols that we create doesn't feel sound. Let's discuss offline in more details.